### PR TITLE
Various minor improvements

### PR DESCRIPTION
**Pull request by xuhcc**
Fixed failing tests and lint errors, improved readme, etc.

### CODE REVIEW

File: contracts/scripts/compileSol.sh [hunk 1]
```diff
@@ -12,7 +12,7 @@ mkdir -p ./compiled/abis
 
 # Copy the Semaphore contracts from the submodule into solidity/
 
-npx etherlime compile --solcVersion=native --buildDirectory=compiled --workingDirectory=sol --exportAbi --runs 200
+npx etherlime compile --solcVersion=0.5.16 --buildDirectory=compiled --workingDirectory=sol --exportAbi --runs 200
```
Comment (weijiekoh) [inline contracts/scripts/compileSol.sh:15]:
```suggestion
if [[ $* == *--native* ]]; then
    npx etherlime compile --solcVersion=native --buildDirectory=compiled --workingDirectory=sol --exportAbi --runs 200
else
    npx etherlime compile --solcVersion=0.5.16 --buildDirectory=compiled --workingDirectory=sol --exportAbi --runs 200
fi
```

I suggest giving the developer the option to use the `native` flag as the solcjs compiler is slower than the solc binary.

File: circuits/scripts/buildBatchUpdateStateTreeSnark.sh [hunk 2]
```diff
@@ -1,10 +1,12 @@
 #!/bin/bash
 
+set -e
+
 cd "$(dirname "$0")"
 cd ..
 mkdir -p build
 
-node build/buildSnarks.js -i circom/prod/batchUpdateStateTreeVerifier.circom -j build/batchUstCircuit.json -p build/batchUstPk.json -v build/batchUstVk.json -s build/BatchUpdateStateTreeVerifier.sol -vs BatchUpdateStateTreeVerifier -m build/batchUst.params -z ~/.cargo/bin/zkutil
+node build/buildSnarks.js -i circom/prod/batchUpdateStateTreeVerifier.circom -j build/batchUstCircuit.json -p build/batchUstPk.json -v build/batchUstVk.json -s build/BatchUpdateStateTreeVerifier.sol -vs BatchUpdateStateTreeVerifier -m build/batchUst.params
```
Comment (weijiekoh) [inline circuits/scripts/buildBatchUpdateStateTreeSnark.sh:9]: The `buildSnarks` command will fail without specifying `-z`.

Comment (weijiekoh) [reply circuits/scripts/buildBatchUpdateStateTreeSnark.sh:9]: Ah, never mind, I see that the default is set in the config!

File: contracts/scripts/compileSol.sh (hunk 1 above)
Comment (xuhcc) [reply contracts/scripts/compileSol.sh]: Ah, sure! I didn't know about that.
Though I'd like to write it differently: 81f85f1af667cb14bfacd058ffd1b8a09e219dad

**Pull request merged**